### PR TITLE
Rid resource group on hashagg spill evaluation

### PR DESCRIPTION
**Pull request by weinan003**
Resource group believe memory access speed always faster than disk,
and it adds hashagg executor node spill mechanism into its memory management.
If the hash table size overwhelms `max_mem`, in resource group model, the hash
table does not spill and fan out data. Resource group wants to grant more memory
for the hash table. However, this strategy impact hash collision rate, so that
some performance regression in some OLAP query.

We rid resource group guc when hashagg evaluate if it needs to spill.

Co-authored-by: Adam Li <ali@pivotal.io>

### CODE REVIEW

File: src/backend/executor/execHHashagg.c [hunk 1]
```diff
@@ -315,22 +315,7 @@ makeHashAggEntryForInput(AggState *aggstate, TupleTableSlot *inputslot, uint32 h
 	if (GET_TOTAL_USED_SIZE(hashtable) + MAXALIGN(MAXALIGN(tup_len) + aggs_len) >=
```
Comment (berlin-ab) [inline src/backend/executor/execHHashagg.c]: A comment above these guard clause validations would help me understand the logic more quickly.

File: src/backend/executor/execHHashagg.c [hunk 2]
```diff
@@ -378,22 +363,7 @@ makeHashAggEntryForGroup(AggState *aggstate, void *tuple_and_aggs,
 
 	if (GET_TOTAL_USED_SIZE(hashtable) + input_size >= hashtable->max_mem)
```
Comment (berlin-ab) [inline src/backend/executor/execHHashagg.c]: see comment above about guard clause validation comments.

File: src/backend/executor/execHHashagg.c [hunk 3]
```diff
@@ -378,22 +363,7 @@ makeHashAggEntryForGroup(AggState *aggstate, void *tuple_and_aggs,
...
-				 "need %d bytes to store the new data; "
-				 "the overuse is allowed in resource group mode",
-				 hashtable->max_mem,
-				 GET_TOTAL_USED_SIZE(hashtable),
-				 input_size);
-		}
-		else
-		{
 			return NULL;
-		}
```
Comment (pivotal-ning-yu) [inline src/backend/executor/execHHashagg.c]: LGTM, but please also adjust the indention of `return NULL;` and remove the `{ }` around it.

**Comment by pivotal-ning-yu** [comment]
This PR is causing a resgroup test failure:
https://prod.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb_master/jobs/resource_group_centos7/builds/821

```diff
--- \/home\/gpadmin\/gpdb_src\/src\/test\/isolation2\/expected\/resgroup\/resgroup_agghash_memory\.out	2019-07-23 06:29:30.647050942 +0000
+++ \/home\/gpadmin\/gpdb_src\/src\/test\/isolation2\/results\/resgroup\/resgroup_agghash_memory\.out	2019-07-23 06:29:30.648051024 +0000
@@ -52,9 +55,9 @@
 SET ROLE TO r1_agghash_mem_test;
 SET
 WITH a AS ( SELECT DISTINCT c2 FROM t1_agghash_mem_test INTERSECT SELECT DISTINCT c2 FROM t1_agghash_mem_test ) SELECT count(*) FROM a;
- count  
---------
- 100000 
+ count 
+-------
+ 23927 
 (1 row)
 RESET role;
 RESET
@@ -64,9 +67,9 @@
 SET ROLE TO r1_agghash_mem_test;
 SET
 WITH a AS ( SELECT DISTINCT c2 FROM t1_agghash_mem_test INTERSECT SELECT DISTINCT c2 FROM t1_agghash_mem_test ) SELECT count(*) FROM a;
- count  
---------
- 100000 
+ count 
+-------
+ 23927 
 (1 row)
 RESET role;
 RESET
```

Would you please take a look?

**Pull request merged**